File: decorator/java/LatteCoffee.java
package io.github.androidhot.decorator.java;

public class LatteCoffee implements Coffee {
  private Coffee coffee;

  public LatteCoffee(Coffee coffee) {
    this.coffee = coffee;
  }

  @Override
  public String name() {
    return coffee.name() + "with Latte ";
  }

  @Override
  public float price() {
    return coffee.price() + 3;
  }
}

File: decorator/kotlin/LatteCoffee.kt
package io.github.androidhot.decorator.kotlin

class LatteCoffee(private val coffee: Coffee) : Coffee {
  override fun name(): String {
    return coffee.name() + "with Latte "
  }

  override fun price(): Float {
    return coffee.price() + 3f
  }
}
